synapse/tests/test_lib_stormlib_task.py

```diff
@@ -0,0 +1,98 @@
+            self.stormIsInPrint('00.cortex.synapse', msgs)
+            self.stormIsInPrint('task.list', msgs)
+            self.stormIsInPrint('01.cortex.synapse', msgs)
+            self.stormIsInPrint('$lib.print(root)', msgs)
+            self.stormIsInPrint('2 tasks', msgs)
+
+            user = await core01.auth.addUser('someuser')
+
+            async with core00.getLocalProxy(user='someuser') as prox00:
+                iden1 = s_common.guid()
+                task1 = aha.schedCoro(prox00.callStorm('$lib.print(on00) $lib.time.sleep(5)', opts={'task': iden1}))
+
+                async with core01.getLocalProxy(user='someuser') as prox01:
+                    msgs = await s_test.alist(prox01.storm('task.list'))
+                    self.stormIsInPrint('00.cortex.synapse', msgs)
+                    self.stormIsInPrint('$lib.print(on00)', msgs)
+                    self.stormIsInPrint('01.cortex.synapse', msgs)
+                    self.stormIsInPrint('task.list', msgs)
+                    self.stormIsInPrint('2 tasks', msgs)
+
+                    await user.addRule((True, ('task', 'get')))
+
+                    msgs = await s_test.alist(prox00.storm('task.list'))
+                    self.stormIsInPrint('4 tasks', msgs)
+
+                    msgs = await s_test.alist(prox01.storm('task.list'))
+                    self.stormIsInPrint('4 tasks', msgs)
+
+                    # We can kill our own  task
+                    with self.raises(s_exc.SynErr):
+                        await prox01.callStorm(f'task.kill {iden1}')
+                        await task1
+
+                    # No task matches
+                    with self.raises(s_exc.StormRuntimeError) as exc:
+                        await prox01.callStorm('task.kill newp')
+                    self.isin('does not match any tasks', exc.exception.get('mesg'))
+
+                    with self.raises(s_exc.StormRuntimeError) as exc:
+                        await prox01.callStorm('task.kill ""')
+                    self.isin('empty task iden prefix', exc.exception.get('mesg'))
+
+                    iden2 = 'c7fc6d4ced5759fe6fcc047b6bd3374b'
+                    task2 = aha.schedCoro(core00.stormlist('$lib.print(root) $lib.time.sleep(5)', opts={'task': iden2}))
+
+                    # Matches exist but we don't have perms to see them
+                    with self.raises(s_exc.StormRuntimeError) as exc:
+                        await prox01.callStorm('task.kill c7fc')
+                    self.isin('does not match any tasks', exc.exception.get('mesg'))
+
+                    await user.addRule((True, ('task', 'del')))
+
+                    # Multiple matches
+                    with self.raises(s_exc.StormRuntimeError) as exc:
+                        await prox01.callStorm('task.kill c7fc')
+                    self.isin('more than one task', exc.exception.get('mesg'))
+
+                    with self.raises(asyncio.CancelledError):
```

Comment:
Are these raising a CancelledError because of timeouts on the scatter-gather API calls? This would be leaking a Python exception through the runtime and be un-catchable by a caller of `$lib.task.kill()`.

If that is the case, we should be raising our own `s_exc.TimeOut` timeout exception instead of leaking the `asyncio.CancelledError`. We **should** be safe to make this change.

Reply:
Re-reading this - if the intent is to show task0 was cancelled, we should only be using the `await task0` statement  under the context manager to show the cancellederror was raised.

Reply:
Updated to move the callStorm out of the context manager